libdnf/repo/solv_repo.cpp

```diff
@@ -697,4 +685,66 @@ std::filesystem::path SolvRepo::solv_file_path(const char * type) {
     return std::filesystem::path(config.get_cachedir()) / CACHE_SOLV_FILES_DIR / solv_file_name(type);
 }
 
+bool SolvRepo::read_group_solvable_from_xml(const std::string & path) {
+    auto & logger = *base->get_logger();
+    solv::Pool & pool = static_cast<solv::Pool &>(get_comps_pool(base));
+    bool read_success = true;
+
+    fs::File ext_file;
+    try {
+        ext_file = fs::File(path, "r", true);
+    } catch (std::filesystem::filesystem_error & e) {
+        logger.warning("Cannot load group extension for system repo from \"{}\": {}", path, e.what());
+        read_success = false;
+    }
+
+    if (read_success) {
+        logger.debug("Loading group extension for system repo from \"{}\"", path);
+        read_success = repo_add_comps(comps_repo, ext_file.get(), 0) == 0;
+        if (read_success) {
+            comps_solvables_end = pool->nsolvables;
```

Comment:
Calling this from `fix_group_missing_xml()` could also mess up the `comps_solvables_end` if there were additional data in the `@system` repo after the group solvables.

Like I mention in the other comment this is currently not an issue.